pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -108,8 +98,9 @@ func normalizeClusters(push *model.PushContext, proxy *model.Proxy, clusters []*
 	return out
 }
 
-func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environment, push *model.PushContext) []*v2.Cluster {
+func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environment, proxy *model.Proxy, push *model.PushContext) []*v2.Cluster {
```

Comment:
with this additional proxy argument, pilot/pkg/networking/core/v1alpha3/configgen.go needs to be changed to add that argument as well. I guess that you'd pass in nil since the caller wouldn't have the proxy handle when it's called.

With the mtls inclusion set, I see the code logic is quite hard to read. Unless it's potentially useful per POD, I don't see the value of adding it.